qiita_db/base.py

```diff
@@ -128,28 +130,34 @@ def _check_id(self, id_):
         the other classes. However, still defining here as there is only one
         subclass that doesn't follow this convention and it can override this.
         """
-        self._check_subclass()
-
-        conn_handler = SQLConnectionHandler()
-
-        return conn_handler.execute_fetchone(
-            "SELECT EXISTS(SELECT * FROM qiita.{0} WHERE "
-            "{0}_id=%s)".format(self._table), (id_, ))[0]
+        sql = "SELECT EXISTS(SELECT * FROM qiita.{0} WHERE {0}_id=%s)".format(
+            self._table)
+        trans.add(sql, [id_])
+        return trans.execute()[-1][0][0]
 
-    def __init__(self, id_):
+    def __init__(self, id_, trans=None):
         r"""Initializes the object
 
         Parameters
         ----------
-        id_: the object identifier
+        id_: object
```

Comment:
Got it, thanks!

On (Jun-30-15|14:45), josenavas wrote:
> >          r"""Initializes the object
> >  
> >          Parameters
> >          ----------
> > -        id_: the object identifier
> > +        id_: object
> 
> Nope, each subclass define which type is it. Usually it is an integer, but there is a couple of cases where it actually is a string; so that's why I have to go back to the common class, which is object.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/1298/files#r33627467